src/SensitiveFloats.jl

```diff
@@ -154,76 +128,61 @@ each evaluated at (sf1, sf2).
 The result is stored in sf_result.  The order is done in such a way that
 it can overwrite sf1 or sf2 and still be accurate.
 """
-function combine_sfs!{ParamType <: ParamSet,
-                      T1 <: Number, T2 <: Number, T3 <: Number}(
-        sf1::SensitiveFloat{ParamType, T1},
-        sf2::SensitiveFloat{ParamType, T1},
-        sf_result::SensitiveFloat{ParamType, T1},
-        v::T1, g_d::Vector{T2}, g_h::Matrix{T3},
-        calculate_hessian::Bool=true)
+function combine_sfs!{T1 <: Number, T2 <: Number, T3 <: Number}(
+                        sf1::SensitiveFloat{T1},
+                        sf2::SensitiveFloat{T1},
+                        sf_result::SensitiveFloat{T1},
+                        v::T1,
+                        g_d::Vector{T2},
+                        g_h::Matrix{T3})
     # You have to do this in the right order to not overwrite needed terms.
-    if calculate_hessian
+    if sf_result.has_hessian
         combine_sfs_hessian!(sf1, sf2, sf_result, g_d, g_h)
```

Comment:
I did find that happening in a couple of tests. I changed those, and adding an assertion to `SensitiveFloat` to ensure that if we're storing/computing a Hessian, we're also storing/computing a gradient. I think we're OK just not allowing users to compute Hessians wo gradients.